File: backend-dotnet/Controllers/CartController.cs
using Microsoft.AspNetCore.Mvc;
using ClothingStore.Data;
using ClothingStore.Models;
using Microsoft.EntityFrameworkCore;

namespace ClothingStore.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class CartController : ControllerBase
    {
        private readonly ClothingStoreContext _context;

        public CartController(ClothingStoreContext context)
        {
            _context = context;
        }

        [HttpGet("user/{userId}")]
        public async Task<ActionResult<List<CartItem>>> GetCartItems(long userId)
        {
            try
            {
                var cartItems = await _context.CartItems
                    .Where(c => c.UserId == userId)
                    .ToListAsync();

                return Ok(cartItems);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpPost("add")]
        public async Task<ActionResult> AddToCart([FromBody] AddToCartRequest request)
        {
            try
            {
                var existingItem = await _context.CartItems
                    .FirstOrDefaultAsync(c => c.UserId == request.UserId && 
                                            c.ProductId == request.ProductId && 
                                            c.Size == request.Size);

                if (existingItem != null)
                {
                    existingItem.Quantity += request.Quantity;
                }
                else
                {
                    var cartItem = new CartItem
                    {
                        UserId = request.UserId,
                        ProductId = request.ProductId,
                        Quantity = request.Quantity,
                        Size = request.Size
                    };
                    _context.CartItems.Add(cartItem);
                }

                await _context.SaveChangesAsync();
                return Ok(new { message = "Item added to cart" });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpDelete("clear/{userId}")]
        public async Task<ActionResult> ClearCart(long userId)
        {
            try
            {
                var cartItems = await _context.CartItems
                    .Where(c => c.UserId == userId)
                    .ToListAsync();

                _context.CartItems.RemoveRange(cartItems);
                await _context.SaveChangesAsync();

                return Ok(new { message = "Cart cleared" });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }
    }

    public class AddToCartRequest
    {
        public long UserId { get; set; }
        public long ProductId { get; set; }
        public int Quantity { get; set; }
        public string? Size { get; set; }
    }
}

File: backend-nodejs/controllers/CartController.js
const CartItem = require('../models/CartItem');
const { Op } = require('sequelize');

class CartController {
  async addToCart(req, res) {
    try {
      const { userId, productId, quantity, size } = req.body;
      
      const existingItem = await CartItem.findOne({
        where: { userId, productId }
      });
      
      if (existingItem) {
        existingItem.quantity += quantity;
        await existingItem.save();
        return res.json(existingItem);
      }
      
      const cartItem = await CartItem.create({
        userId,
        productId,
        quantity,
        size
      });
      
      res.json(cartItem);
    } catch (error) {
      res.status(400).json({ error: error.message });
    }
  }

  async getCartItems(req, res) {
    try {
      const { userId } = req.params;
      const cartItems = await CartItem.findAll({
        where: { userId }
      });
      res.json(cartItems);
    } catch (error) {
      res.status(500).json({ error: error.message });
    }
  }

  async updateQuantity(req, res) {
    try {
      const { userId, productId, quantity } = req.body;
      
      const cartItem = await CartItem.findOne({
        where: { userId, productId }
      });
      
      if (!cartItem) {
        return res.status(404).json({ error: 'Cart item not found' });
      }
      
      cartItem.quantity = quantity;
      await cartItem.save();
      
      res.json(cartItem);
    } catch (error) {
      res.status(400).json({ error: error.message });
    }
  }

  async removeFromCart(req, res) {
    try {
      const { productId } = req.params;
      const { userId } = req.query;
      
      await CartItem.destroy({
        where: { userId, productId }
      });
      
      res.json({ message: 'Item removed from cart' });
    } catch (error) {
      res.status(400).json({ error: error.message });
    }
  }

  async clearCart(req, res) {
    try {
      const { userId } = req.params;
      
      await CartItem.destroy({
        where: { userId }
      });
      
      res.json({ message: 'Cart cleared' });
    } catch (error) {
      res.status(400).json({ error: error.message });
    }
  }
}

module.exports = new CartController();